src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -393,40 +397,44 @@ public String getStringByName(String memberName) {
     public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
             @Nonnull Launcher launcher, @Nonnull TaskListener listener)
             throws InterruptedException, IOException {        
-        final boolean isLinuxLauncher = launcher.isUnix();
+        //Set the environment variable specific to the this build
+        setEnv(build.getEnvironment(listener));
+        nodeSpecificfileSeparator = getNodeSpecificFileSeperator(launcher);
         
         // Invoke MATLAB command and transfer output to standard
         // Output Console
 
-        buildResult = execMatlabCommand(build, workspace, launcher, listener, isLinuxLauncher);
+        buildResult = execMatlabCommand(workspace, launcher, listener);
 
         if (buildResult != 0) {
             build.setResult(Result.FAILURE);
         }
     }
 
-    private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace, Launcher launcher,
-            TaskListener listener, boolean isLinuxLauncher)
+    private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
+            TaskListener listener)
             throws IOException, InterruptedException {
-        setEnv(build.getEnvironment(listener));
-        final String testRunMode = this.getTestRunTypeList().getDescriptor().getId();
-        
-        // Copy MATLAB scratch file into the workspace only if Automatic option is selected.
-        if (testRunMode.contains(AUTOMATIC_OPTION)) {
-            copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE,
-                    workspace, getClass().getClassLoader());
-        }
         ProcStarter matlabLauncher;
         try {
-            MatlabReleaseInfo rel = new MatlabReleaseInfo(getLocalMatlab());
+            FilePath nodeSpecificMatlabRoot = new FilePath(launcher.getChannel(),getLocalMatlab());
+            MatlabReleaseInfo rel = new MatlabReleaseInfo(nodeSpecificMatlabRoot);
             matlabLauncher = launcher.launch().pwd(workspace).envs(this.env);
             if (rel.verLessThan(BASE_MATLAB_VERSION_BATCH_SUPPORT)) {
                 ListenerLogDecorator outStream = new ListenerLogDecorator(listener);
-                matlabLauncher = matlabLauncher.cmds(constructDefaultMatlabCommand(isLinuxLauncher)).stderr(outStream);
+                matlabLauncher = matlabLauncher.cmds(constructDefaultMatlabCommand(launcher.isUnix())).stderr(outStream);
             } else {
                 matlabLauncher = matlabLauncher.cmds(constructMatlabCommandWithBatch()).stdout(listener);
             }
-        } catch (MatlabVersionNotFoundException e) {
+            
+            //Check the test run mode option selected by user and identify the target workspace to copy the scratch file.
+            final String testRunMode = this.getTestRunTypeList().getDescriptor().getId();
+            FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
```

Comment:
construction of targetworkspace should be done only in the scope that is needed

Reply:
Done.